fuzz/__main__.py

```diff
@@ -13,83 +13,129 @@
-from unittest.mock import patch, MagicMock
 
-cred_sweeper = credsweeper.app.CredSweeper()
+from credsweeper import CredSweeper, DataContentProvider, ApplyValidation
+from credsweeper.validations import GithubTokenValidation, GoogleApiKeyValidation, MailChimpKeyValidation, \
+    StripeApiKeyValidation, SquareClientIdValidation, SlackTokenValidation, SquareAccessTokenValidation, \
+    GoogleMultiValidation
+
+# set log level for fuzzing
+logging.basicConfig(level=logging.CRITICAL)
+logger = logging.getLogger(__name__)
+
+# need to set depth=1 to remove .zip and .gz from extension
+cred_sweeper = CredSweeper(depth=1)
+api_validation = ApplyValidation()
+
+INPUT_DATA_SIZE = 0x0600
+BEHAVIOUR_BYTE_SIZE = 0x01
+MOCK_RESPONSE_SIZE = 0x01FF
+
+
+def mock_request(behaviour_code: int, status_code_seed: int, content: bytes, candidate, patch_object, path_name):
+    if 0 == behaviour_code:
+        response = Response()
+        response._content = content
+        status_codes = [200, 300, 400, 401, 403, 500, 0, 999]
+        # maximum 16 items due only 4 bits is used from mock_fuzz_byte
+        assert len(status_codes) <= 0x10
+        response.status_code = status_codes[status_code_seed % len(status_codes)]
+        logger.debug("<<<<<<<< %d '%s'", response.status_code, content.decode(encoding='ascii', errors='ignore'))
+        with patch.object(patch_object, path_name, return_value=response):
+            api_validation.validate(candidate)
+    if 1 == behaviour_code:
+        # generate common exception
+        logger.debug("<<<<<<<< Side_effect=Exception('fuzz %s Exception')", path_name)
+        with patch.object(patch_object, path_name, side_effect=Exception(f"fuzz {path_name} Exception")):
+            api_validation.validate(candidate)
+    else:
+        # generate ConnectError exception
+        logger.debug("<<<<<<<< %s side_effect=requests.exceptions.ConnectionError", path_name)
+        with patch.object(patch_object, path_name, side_effect=requests.exceptions.ConnectionError):
+            api_validation.validate(candidate)
+
+
+def mock_flow(behaviour_code: int, candidate):
+    if 0 == behaviour_code:
+        logger.debug(f"<<<<<<<< flow.fetch_token.return_value = None")
+        with patch.object(InstalledAppFlow, InstalledAppFlow.from_client_config.__name__) as mock:
+            flow = MagicMock()
+            flow.fetch_token.return_value = None
+            mock.return_value = flow
+            api_validation.validate(candidate)
+    if 1 == behaviour_code:
+        logger.debug(f"<<<<<<<< InvalidGrantError('fuzz InvalidGrantError')")
+        with patch.object(InstalledAppFlow, InstalledAppFlow.from_client_config.__name__) as mock:
+            flow = MagicMock()
+            flow.fetch_token.side_effect = InvalidGrantError('fuzz InvalidGrantError')
+            mock.return_value = flow
+            api_validation.validate(candidate)
+    else:
+        logger.debug(f"<<<<<<<< Exception('fuzz flow Exception')")
```

Comment:
Maybe replace `<<<<<<<<` by some kind of tag? 
It looks kinda messy

Reply:
Probably.
Otherwise, simple graphics is easiest to find than text recognition by human (IMHO). There were many text in debug, but this samples did not appear in data. So, it was choose like a marker to distinguish in(>>>>>) and out(<<<<<).